### DISCUSSION TITLE
Using JavaMelody with Quarkus - any plans!?

### DISCUSSION THREAD

**Discussion by FredrikAnderson** [javamelody/javamelody#1085] [General]
Hi,

Using JavaMelody in many of my apps, for example using Spring Boot with very good experience for first quick metrics and support help in an infrastructure environment.
Now starting to look into Quarkus, using undertow, as a framework and tried a concept of using JavaMelody there too, but ended up with: 
Caused by: javax.enterprise.inject.spi.DefinitionException: Interceptor has no bindings: net.bull.javamelody.MonitoringAsynchronousCdiInterceptor
 
Is there any plans and conceptual tries with JavaMelody for integration with Quarkus (quarkus.io) ?

Thanks for good piece of software and looking forward for response,
Fredrik

**Comment by evernat** [comment]
Hi @FredrikAnderson 
The plan is that it is compatible if possible and make it simple as much as possible. There are no tests with Quarkus underway, except yours. There is no documentation for javamelody with Qarkus and no Quarkus extension planned as I know of, except if you want to write it.

For the exception here, what is your javamelody version and what artifact is used (javamelody-core, javamelody-spring-boot-starter) ?
The version used is to be sure of the content of META-INF/beans.xml in javamelody. The latest javamelody version is 1.90.0.

Then, for sure CDI in your Quarkus version or in your Undertow version does not like the [MonitoringAsynchronousCdiInterceptor](https://github.com/javamelody/javamelody/blob/master/javamelody-core/src/main/java/net/bull/javamelody/MonitoringAsynchronousCdiInterceptor.java) class. The thing is that this javamelody class to monitor beans in CDI having an `javax.ejb.Asynchronous` annotation should be used only when you configure CDI in your app to use this interceptor. So you may find some configuration with CDI in Quarkus / undertow to ignore it, instead of trying to do something with it for a reason I don't know.

**Comment by FredrikAnderson** [comment]
Hi @evernat 
Thanks for your answer and supportive directions.

Was/is playing around with Quarkus and using javamelody-core version 1.90.0.

Now have some steps to get JavaMelody going in Quarkus! Or at least, can browse URL "/monitoring" and see JVM stats, properties, http requests as well SQL statements. Have been able to do this by some quarkus configurations both for ignoring interceptor as you where into and for datasource setup needed a few lines to support that.
Willing to share this steps and information on how I managed and also for someone to possibly verify and give feedback. Please propose some way for sharing information if it is of interest.

**Reply by evernat** [reply]
Hi @FredrikAnderson 
To share steps and information, you can send an email to me: evernat at free dot fr, in English.
I will review/give feedback if needed and probably put that in the javamelody wiki.
We could then send a link in https://groups.google.com/g/javamelody and in https://twitter.com/java_melody to reach people which can also give feedback.

**Reply by evernat** [reply]
or you could write a blog where you want and which could be copied in javamelody wiki

**Reply by evernat** [reply]
blog copied to https://github.com/javamelody/javamelody/wiki/Quarkus
thanks

**Comment by teixi** [comment]
Current quarkus 3.29

quarkus dependencies:
  'io.quarkus:quarkus-rest-jackson'
  'io.quarkus:quarkus-rest-servlet'
 
 javamelody dependencies added via configuration:
  quarkus.index-dependency
  
 configuration:
  quarkus.arc.exclude-types=net.bull.javamelody.MonitoringAsynchronousCdiInterceptor,sun.awt.**
